### ISSUE TITLE
Exchange.publish() is broken when string provided as message

### ISSUE THREAD

**Issue by bernd-wechner** [celery/kombu#1174]
In` Exchange.publish()`, if a string is provided as a message and a content_encoding is not, then it crashes when it comes to `self.channel.prepare_message()`  which has a default value for an explictly named kwarg of None.

```
    def prepare_message(self, body, priority=None,
                        content_type=None, content_encoding=None,
                        headers=None, properties=None):
```
 It subsequently crashes when trying to encode the message.

A number of solutions are possible but the easiest is to assume a string encoding in Echange.Message or use the [chardet](https://pypi.org/project/chardet/) library to estimate one.

The Message method has a lead in comment of:

```
# XXX This method is unused by kombu itself AFAICT [ask].
```

Well I'm asking ;-). It **IS** used by the Exchange.publish() method. To wit should be maintained and fixed.

**Comment by bernd-wechner** [comment]
One proposed PR: https://github.com/celery/kombu/pull/1175

**Comment by bernd-wechner** [comment]
Fix #1174 accepted.
